.github/workflows/deploy_release.yaml

```diff
@@ -75,10 +89,26 @@ jobs:
         env:
           SONATYPE_TOKEN_USERNAME: ${{ vars.KROXYLICIOUS_SONATYPE_TOKEN_USERNAME }}
           SONATYPE_TOKEN_PASSWORD: ${{ secrets.KROXYLICIOUS_SONATYPE_TOKEN_PASSWORD }}
-        run: ./scripts/transition-staging-repository-state.sh -s ${{ github.event.inputs.next-state }}
+        run: ./scripts/transition-staging-repository-state.sh -s ${{ inputs.next-state }}
 
       - name: 'Transition release notes to desired state'
         env:
           GH_TOKEN: ${{ secrets.KROXYLICIOUS_RELEASE_TOKEN }} # For the gh cmd line tool used by transition-github-release-note-state.sh
-        run: ./scripts/transition-github-release-note-state.sh -s  ${{ github.event.inputs.next-state }} -v ${{ github.event.inputs.release-version }}
+        run: ./scripts/transition-github-release-note-state.sh -s  ${{ inputs.next-state }} -v ${{ env.RELEASE_VERSION }}
+
+      - name: 'Merge PR'
+        if: inputs.next-state == 'release'
+        env:
+          GH_TOKEN: ${{ secrets.KROXYLICIOUS_RELEASE_TOKEN }}
+        run: |
+          # TODO: verify that main hasn't moved
+          echo gh pr merge --delete-branch --rebase ${{ inputs.release-pr-issue-number }}
```

Comment:
I want to fail here if main has move.  I am thinking of using `--match-head-commit` with a SHA calculated earlier in the workflow.